### PR TITLE
cellCamera: Logging improvements and hotfix cellCameraInit

### PR DESCRIPTION
**Pull request by elad335**

### CODE REVIEW

**Comment by Megamouse** [comment]
Shouldn't check_camera_info also return error code? Or does s32 get handled like error_coe too?

**Comment by elad335** [comment]
Fixed similar s32 issues as well.

**Comment by elad335** [comment]
Latest commit may fix some regressions, worth testing out.

File: rpcs3/Emu/Cell/Modules/cellCamera.cpp [hunk 1]
```diff
@@ -307,7 +307,7 @@ error_code cellCameraInit()
 
 	std::lock_guard lock(g_camera->mutex);
 
-	if (g_camera->init)
+	if (g_camera->init.exchange(1))
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellCamera.cpp]: this doesnt make any sense.
the init should be set set at the end and inside the null handler return scope

**Comment by Megamouse** [comment]
Btw (unrelated to your changes): Returning CELL_OK with null handlers *before* all the param checks etc. seems like a really bad idea. The games expect the checks to happen if a camera is used or not. So skipping the checks will most likely change the games' main loops and may therefore result in random unexpected behaviour. Imo the null handler should really just act like there wasn't any camera attached.

**Comment by elad335** [comment]
well, It's a bug.

**Comment by Megamouse** [comment]
Does error_code CELL_OK really resolve to false?

**Comment by elad335** [comment]
yes.

**Pull request merged**